lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -776,13 +785,29 @@ bool ProxyRenderDelegate::getInstancedSelectionPath(
     }
 
     const Ufe::PathSegment pathSegment(usdPath.GetText(), USD_UFE_RUNTIME_ID, USD_UFE_SEPARATOR);
-    const Ufe::SceneItem::Ptr& si
-        = handler->createItem(_proxyShapeData->ProxyShape()->ufePath() + pathSegment);
+    Ufe::Path sceneItemPath(_proxyShapeData->ProxyShape()->ufePath() + pathSegment);
+
+    if (instanceIndex != UsdImagingDelegate::ALL_INSTANCES) {
+        const std::string& primName = sceneItemPath.back().string();
+        const Ufe::PathComponent primInstanceComponent(
+            TfStringPrintf("%s.instance_%d", primName.c_str(), instanceIndex));
+        sceneItemPath = sceneItemPath.sibling(primInstanceComponent);
+    }
+
+    const Ufe::SceneItem::Ptr& si = handler->createItem(sceneItemPath);
     if (!si) {
         TF_WARN("Failed to create UFE scene item for Rprim '%s'", rprimId.GetText());
         return false;
     }
 
+    auto usdItem = std::dynamic_pointer_cast<MayaUsd::ufe::UsdSceneItem>(si);
+    if (!usdItem) {
+        TF_WARN("Failed to cast UFE scene item to UsdSceneItem for Rprim '%s'", rprimId.GetText());
+        return false;
+    }
+
+    usdItem->setInstanceIndex(instanceIndex);
```

Comment:
We'll need this value both in `PopulateSelection()` here in the render delegate to make sure the right instance is put in the Hydra selection, but also later when creating the transform handler for the `PointInstancer`, since this will be the index into the various array attributes for the `positions`/`orientations`/`scales`/etc. that we'll need to modify.

For performance, I think we'll really want to keep this available somewhere as an `int` and not have to do any string manipulation to try to yank it out of the `Ufe::Path`. Selecting a large swatch of instances with a marquee selection and then manipulating them all as a group could be a pretty common use case.

Reply:
What is an instance index in the USD data model?  Is it a property of a prim?